code/game/gamemodes/shadowling/shadow_abilities.dm

```diff
@@ -0,0 +1,542 @@
+						user.visible_message("<span class='warning'>[user] pauses, then dips their head in concentration!</span>")
+						to_chat(target, "<span class='boldannounce'>You feel your mental protection faltering</span>")
+						if(!do_mob(user, target, 650)) //65 seconds to remove a loyalty implant. yikes!
+							to_chat(user, "<span class='warning'>The enthralling has been interrupted - your target's mind returns to its previous state.</span>")
+							to_chat(target, "<span class='userdanger'>You wrest yourself away from [user]'s hands and compose yourself</span>")
+							enthralling = 0
+							return
+						to_chat(user, "<span class='notice'>The nanobots composing the mindshield implant have been rendered inert. Now to continue.</span>")
+						user.visible_message("<span class='warning'>[user] relaxes again.</span>")
+						for(var/obj/item/implant/mindshield/L in target)
+							if(L)
+								qdel(L)
+						to_chat(target, "<span class='boldannounce'>Your mental protection unexpectedly falters, dims, dies.</span>")
+				if(3)
+					to_chat(user, "<span class='notice'>You begin planting the tumor that will control the new thrall...</span>")
+					user.visible_message("<span class='warning'>A strange energy passes from [user]'s hands into [target]'s head!</span>")
+					to_chat(target, "<span class='boldannounce'>You feel your memories twisting, morphing. A sense of horror dominates your mind.</span>")
+			if(!do_mob(user, target, 70)) //around 21 seconds total for enthralling, 86 for someone with a loyalty implant
+				to_chat(user, "<span class='warning'>The enthralling has been interrupted - your target's mind returns to its previous state.</span>")
+				to_chat(target, "<span class='userdanger'>You wrest yourself away from [user]'s hands and compose yourself</span>")
+				enthralling = FALSE
+				return
+
+		enthralling = FALSE
+		to_chat(user, "<span class='shadowling'>You have enthralled <b>[target.real_name]</b>!</span>")
+		target.visible_message("<span class='big'>[target] looks to have experienced a revelation!</span>", \
+							   "<span class='warning'>False faces all d<b>ark not real not real not--</b></span>")
+		target.setOxyLoss(0) //In case the shadowling was choking them out
+		target.mind.special_role = "thrall"
+		var/obj/item/organ/internal/shadowtumor/ST = new
+		ST.Insert(target, FALSE, FALSE)
+		add_thrall(target.mind)
+		if(target.reagents.has_reagent("frostoil")) //Stabilize body temp incase the sling froze them earlier
+			target.reagents.remove_reagent("frostoil")
+			to_chat(target, "<span class='notice'>You feel warmer... it feels good.</span>")
+			target.bodytemperature = 310
+
+
+
+
+//////////////////////
+////SLING HIVEMIND////
+//////////////////////
+
+/obj/effect/proc_holder/spell/shadowling_hivemind //Lets a shadowling talk to its allies
+	name = "Hivemind Commune"
+	desc = "Allows you to silently communicate with all other shadowlings and thralls."
```

Comment:
This is probably best refactored using language systems so you can put a `.b` or something before your message and it'll send to the Hivemind. I was able to do it for umbrages when I was working on them, so it's absolutely doable.

Reply:
Saymode datums?